spyder/utils/programs.py

```diff
@@ -462,6 +462,24 @@ def is_module_installed(module_name, version=None, installed_version=None,
             
             return check_version(actver, version, symb)
 
+def is_python_interpreter(filename):
+    """Evaluate wether a file is a python interpreter or not."""
+    real_filename = os.path.realpath(filename) # To follow symlink if existent
+    if not osp.isfile(real_filename) or encoding.is_text_file(real_filename):
+        return False
+    possible_names = ['python', 'python2', 'python2.7', 'python3', 'python3.4',
+                      'python3.5']

```

Comment:
Maybe using a regex is better

`python(?P<version>\d\.?\d*)?$`

This will match any of the options plus any version.. then we can check that version part (if not empty) is a float